### PR TITLE
Cancel gift aid

### PR DESCRIPTION
**Pull request by ivan-unfolds**
@rjmk I decided to do the changes the accompany cancelling the Gift Aid declaration on the front-end as they did not depend on any previously stored information. What do you think?

Also I am unsure if I need to change all the fields that I did. Maybe `gift_aid_signed` and `date_gift_aid_signed` need to remain as they are? Otherwise I don't see the point of having both `gift_aid_signed` and `gift_aid_cancelled`.

### CODE REVIEW

File: src/shared/components/payments_table/deletion_entry.js [hunk 1]
```diff
@@ -1,25 +1,35 @@
-var request = require('xhr')
 var React = require('react')
 
 module.exports = React.createClass({
   getInitialState: function () {
-    return { confirmation: false } },
+    return { confirmation: false }
+  },
```
Comment (rjmk) [inline src/shared/components/payments_table/deletion_entry.js]: Please don't do this mixed in with a PR that has some stuff going on. It makes the diff very noisy

File: src/styles/components/_button.scss [hunk 2]
```diff
@@ -83,7 +83,14 @@ button {
 	background-color: #ffe5e5;
 	&:active, &:focus {
 		background-color: #ffe5e5;
-	}	
+	}
+}
+
+.green {
+	background-color: #a7dc95;
```
Comment (rjmk) [inline src/styles/components/_button.scss:90]: I think you've got some tabs here

File: src/user/components/my_details/cancel_gift_aid_button.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+export default class GiftAidButton extends React.Component {
```
Comment (rjmk) [inline src/user/components/my_details/cancel_gift_aid_button.js]: Let's stick with using `React.createClass` as that's what's used elsewhere

File: src/user/components/my_details/cancel_gift_aid_button.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import React from 'react'
```
Comment (rjmk) [inline src/user/components/my_details/cancel_gift_aid_button.js]: Can we reduce the duplication between this and `deletion_entry.js`?

File: src/user/components/my_details/cancel_gift_aid_button.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+export default class GiftAidButton extends React.Component {
+  constructor (props) {
+    super (props)
+    this.state = { confirmation: false }
+  }
+
+  cancel () {
+    this.props.update_member_user(
```
Comment (rjmk) [inline src/user/components/my_details/cancel_gift_aid_button.js]: I don't think we want all this transformation of the user down here -- it should be handled in the reducer that knows about the shape of the state

**Comment by rjmk** [comment]
Yep, I think keep those two variables as they are. Really there shouldn't be both a `gift_aid_signed` and a `gift_aid_cancelled` in the db as that gives 4 possibilities but only 3 of them have meaning. Oh well!

File: src/shared/components/payments_table/deletion_entry.js (hunk 1 above)
Comment (ivan-unfolds) [reply src/shared/components/payments_table/deletion_entry.js]: Sorry!

File: src/user/components/my_details/cancel_gift_aid_button.js (hunk 4 above)
Comment (ivan-unfolds) [reply src/user/components/my_details/cancel_gift_aid_button.js]: Will attempt this now.

File: src/styles/components/_button.scss (hunk 2 above)
Comment (ivan-unfolds) [reply src/styles/components/_button.scss:90]: Not sure what you mean.

Comment (rjmk) [reply src/styles/components/_button.scss:90]: There are hard tab characters here

Looks like the whole file has them though, so nevermind

**Comment by ivan-unfolds** [comment]
@rjmk I am unsure about doing the transformation of the user in the reducer as you suggested. As I understand it that would mean passing down another field into the redux form, creating an action to look out for this change and then sending the form off. The way I have currently done it, the cancel Gift Aid button does not have anything to do with the redux form. Therefore you do not need to enter 'edit' mode, which is a prerequisite for sending the form off unless I am mistaken. I think the UI works better this way.

Also at the moment this cancel gift aid buttons are only displayed if `gift_aid_cancelled` is false. Pressing this button only changes this entry in DB (as well as `date_gift_aid_cancelled`). After that the button is not displayed anymore but the state is updated with new values from response. Therefore I do not see the benefits of doing these changes in reducer. What's your opinion?

**Comment by rjmk** [comment]
@ivanmauricio I would stick with doing it in the reducer. It doesn't have to be done through Redux Form. Have a look at something like `DEACTIVATED_MEMBER`.

I definitely agree that this is the UI we want. But the UI should not contain the logic for updating the state

**Comment by ivan-unfolds** [comment]
@rjmk For `DEACTIVATED_MEMBER` the delete button changes this field in the state but it does not get updated on the DB. It only gets updated when you press save, triggering a form submission. This is why I thought that I would have to involve redux form. Is that not the case?

**Comment by ivan-unfolds** [comment]
@rjmk Also I sent you a message on gitter.

**Comment by rjmk** [comment]
@ivanmauricio Sorry I was slightly wrong in my comment -- the changes to the state of the member should not happen in the reducer, but in the action creator. You should export a specific action for this

**Comment by rjmk** [comment]
@ivanmauricio Will check gitter

**Pull request merged**